crabs/detection_tracking/inference_model.py

```diff
@@ -136,50 +153,173 @@ def prep_csv_writer(self):
+        tracked_boxes = self.sort_tracker.update(pred_sort)
+        self.tracked_list.append(tracked_boxes)
+        return tracked_boxes
+
+    def handle_output(
+        self,
+        tracked_boxes: List[List[float]],
+        frame: np.ndarray,
+        frame_number: int,
+    ) -> None:
+        """
+        Handle the output based argument options.
+
+        Parameters:
+        -----------
+        tracked_boxes : List[List[float]]
+            List of tracked bounding boxes.
+        frame : np.ndarray
+            The current frame.
+        frame_number : int
+            The frame number.
+        """
+        if self.args.save_csv_and_frames:
+            if self.args.save_video:
+                frame_copy = save_frame_and_csv(
+                    self.video_file_root,
+                    self.tracking_output_dir,
+                    tracked_boxes,
+                    frame,
+                    frame_number,
+                    self.csv_writer,
+                )
+                for bbox in tracked_boxes:
+                    xmin, ymin, xmax, ymax, id = bbox
+                    draw_bbox(
+                        frame_copy,
+                        int(xmin),
+                        int(ymin),
+                        int(xmax),
+                        int(ymax),
+                        (0, 0, 255),
+                        f"id : {int(id)}",
+                    )
+                self.out.write(frame_copy)
+            else:
+                save_frame_and_csv(
+                    self.video_file_root,
+                    self.tracking_output_dir,
+                    tracked_boxes,
+                    frame,
+                    frame_number,
+                    self.csv_writer,
+                    save_plot=False,
+                )
+        elif self.args.save_video:
+            frame_copy = frame.copy()
+            for bbox in tracked_boxes:
+                xmin, ymin, xmax, ymax, id = bbox
+                draw_bbox(
+                    frame_copy,
+                    int(xmin),
+                    int(ymin),
+                    int(xmax),
+                    int(ymax),
+                    (0, 0, 255),
+                    f"id : {int(id)}",
+                )
+            self.out.write(frame_copy)
 
     def run_inference(self):
         """
         Run object detection + tracking on the video frames.
         """
         # Get transform to tensor
-        transform = transforms.Compose([transforms.ToTensor()])
+        transforms.Compose([transforms.ToTensor()])
 
-        # initialise frame counter
+        # initialisation
         frame_number = 1
+        self.tracked_list = []
 
         # initialise csv writer if required
         if self.args.save_csv_and_frames:
-            csv_writer, csv_file = self.prep_csv_writer()
+            self.csv_writer, csv_file = self.prep_csv_writer()
+
+        if self.args.gt_dir:
+            from crabs.detection_tracking.detection_utils import (
```

Comment:
I think I would also have this import at the top, even if only used in this case.

Also, I think we can move the lien with `gt_boxes_list = get_ground_truth_data(self.args.gt_dir)` to be under the next `if self.args.gt_dir:` and we save an if-case right?